pkg/reconciler/buildrun/resources/image_processing_test.go

```diff
@@ -0,0 +1,185 @@
+// Copyright The Shipwright Contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package resources_test
+
+import (
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	corev1 "k8s.io/api/core/v1"
+
+	buildv1alpha1 "github.com/shipwright-io/build/pkg/apis/build/v1alpha1"
+	"github.com/shipwright-io/build/pkg/config"
+	"github.com/shipwright-io/build/pkg/reconciler/buildrun/resources"
+	"github.com/shipwright-io/build/test/utils"
+	pipeline "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+)
+
+var _ = Describe("Image Processing overrides", func() {
+
+	config := config.NewDefaultConfig()
+	var processedTaskRun *pipeline.TaskRun
+
+	Context("for a TaskRun that does not reference the output directory", func() {
+		taskRun := &pipeline.TaskRun{
+			Spec: pipeline.TaskRunSpec{
+				TaskSpec: &pipeline.TaskSpec{
+					Steps: []pipeline.Step{
+						{
+							Name: "test-step",
+						},
+					},
+				},
+			},
+		}
+
+		Context("for a build without labels and annotation in the output", func() {
+			BeforeEach(func() {
+				processedTaskRun = taskRun.DeepCopy()
+				resources.SetupImageProcessing(processedTaskRun, config, buildv1alpha1.Image{
+					Image: "some-registry/some-namespace/some-image",
+				}, buildv1alpha1.Image{})
+			})
+
+			It("does not add the image-processing step", func() {
+				Expect(processedTaskRun.Spec.TaskSpec.Steps).To(HaveLen(1))
+				Expect(processedTaskRun.Spec.TaskSpec.Steps).ToNot(utils.ContainNamedElement("image-processing"))
+			})
+		})
+
+		Context("for a build with a label in the output", func() {
+			BeforeEach(func() {
+				processedTaskRun = taskRun.DeepCopy()
+				resources.SetupImageProcessing(processedTaskRun, config, buildv1alpha1.Image{
+					Image: "some-registry/some-namespace/some-image",
+					Labels: map[string]string{
+						"aKey": "aLabel",
+					},
+				}, buildv1alpha1.Image{})
+			})
+
+			It("adds the image-processing step", func() {
+				Expect(processedTaskRun.Spec.TaskSpec.Steps).To(HaveLen(2))
+				Expect(processedTaskRun.Spec.TaskSpec.Steps[1].Name).To(Equal("image-processing"))
+				Expect(processedTaskRun.Spec.TaskSpec.Steps[1].Image).To(Equal(config.ImageProcessingContainerTemplate.Image))
+				Expect(processedTaskRun.Spec.TaskSpec.Steps[1].Command).To(Equal(config.ImageProcessingContainerTemplate.Command))
+				Expect(processedTaskRun.Spec.TaskSpec.Steps[1].Args).To(Equal([]string{
+					"--label",
+					"aKey=aLabel",
+					"--image",
+					"$(params.shp-output-image)",
+					"--insecure=$(params.shp-output-insecure)",
+					"--result-file-image-digest",
+					"$(results.shp-image-digest.path)",
+					"--result-file-image-size",
+					"$(results.shp-image-size.path)",
+				}))
+				Expect(processedTaskRun.Spec.TaskSpec.Steps[1].VolumeMounts).ToNot(utils.ContainNamedElement("shp-output-directory"))
+			})
+		})
+	})
+
+	Context("for a TaskRun that references the output directory", func() {
```

Comment:
don't u need also the one where the label is also set(together with output directory)?

Reply:
Added.